flatdata-cpp/include/flatdata/ExternalVector.h

```diff
@@ -97,17 +103,13 @@ ExternalVector< T >::flush( )
 }
 
 template < typename T >
-boost::optional< ArrayView< T > >
+ArrayView< T >
 ExternalVector< T >::close( )
 {
     flush( );
-    boost::optional< MemoryDescriptor > data = m_array->close( );
-    if ( !data )
-    {
-        return boost::none;
-    }
-
-    return ArrayView< T >{data->data( ), data->data( ) + data->size_in_bytes( )};
+    MemoryDescriptor data = m_array->close( );
+    return data ? ArrayView< T >{data.data( ), data.data( ) + data.size_in_bytes( )}
```

Comment:
Actually no need to check, it will just work magically :-)

Reply:
Yeah! I rather decided not to look up nullptr arithmetic.